### DISCUSSION TITLE
Channels Stuck Closing

### DISCUSSION THREAD

**Discussion by pasq1** [lightningnetwork/lnd#8134] [Troubleshooting]
I recovered my node and most of my channels recovered/restored the funds to on-chain besides 2 of them. They have been stuck closing for about a month now and I've tried to open other channels and add the other nodes as peers but those 2 specifically wont let me connect. I'm all out of ideas at this point any tips?

Ive tried to look at the pending channels command through CLI and I receive this is return
"{
    "total_limbo_balance": "0",
    "pending_open_channels": [
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "028742b6b8eb7f28890cb232b2d490f007baf9ced2af1519f61413ba0bdae32889",
                "channel_point": "b75c631cd20c6c18e62a1f4d667cfdc950503ba6149e3b3afa7653bc76225909:0",
                "capacity": "1000000",
                "local_balance": "0",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusRestored",
                "private": true
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "",
                "remote_txid": "",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "0",
                "remote_commit_fee_sat": "0",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "02ab1fbe10650d06f0b94fb4f61d7c4d189b8b0a9f818ea1a9b49667e95a768821",
                "channel_point": "d49bc11a7c8764b47ed9185e8dd1c6a9a5e513c5c09672d483a6c1c141d1ba1e:1",
                "capacity": "1000000",
                "local_balance": "0",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusRestored",
                "private": true
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "",
                "remote_txid": "",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "0",
                "remote_commit_fee_sat": "0",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        }
    ]
}"

Thanks for any suggestions!

**Comment by bitromortac** [comment]
These are so-called zombie channels, i.e., your peers are offline and you therefore you can't get them to force close with you via the `option_data_loss_protect` protocol (more info here https://docs.lightning.engineering/lightning-network-tools/lnd/safety#zombie-channels). I would recommend to start over with the node and make an entry in https://www.node-recovery.com/ just in case the other peers look for you (and they need to recover as well). Should the peers come online again and force close by themselves then you can recover those funds. So it would probably make sense to keep that node running somewhere. Did you suffer from a complete data loss?

**Comment by pasq1** [comment]
I'm sorry I don't know exactly what you mean by complete data loss. My node wouldn't stay synced with the chain so I redownloaded all the software and resynced the chain. After that I used the seed and backup file from the old instance, that closed all my channels except these two. I don't know if that counts as a complete data loss or not. Just to be clear your saying I should close any existing channels I have and re-download the node software and try to recover again along with posting on the node-recovery website? Also I'm pretty sure I tried force closing the channels and it said that it couldn't because they were restored channels. Thanks a ton

**Reply by bitromortac** [reply]
Do you use any node packaging software, like Umbrel?

**Reply by pasq1** [reply]
Yes Umbrel with just LND

**Reply by bitromortac** [reply]
Ok, with reinstalling LND on Umbrel important data (`channel.db`) gets lost (at least that's the latest state of my knowledge). This is equivalent of suffering from a data loss. The static channel backup was designed for these situations, but you can't restore all funds in every case unfortunately (which is why "backup" is a misnomer). You need to rely on your peers to force close those channels, but if they are offline, these channels stay open until they get closed by your peers. Reinstalling the node software is not good practice, instead of reinstalling you would have needed to force close yourself before this process. This is not possible anymore unfortunately because `channel.db` is gone. My understanding here could be incorrect in terms of what gets deleted exactly, so I'd recommend to ask in the Umbrel telegram channel to confirm.

**Reply by pasq1** [reply]
Thats the funny part I've gone back and forth with the Umbrel Support team for a month and they said to make a post here for more indepth lightning help, what would you like me to ask them specially?

**Reply by bitromortac** [reply]
I would ask them if channel.db gets deleted upon uninstallation and reinstallation.

**Reply by pasq1** [reply]
Just did ill let you know the response, with the zombie channels is there anything I can do about it or is it just in purgatory because the other nodes see offline? (Amboss says they are active but when I try to add them is says failed to connect)

**Reply by bitromortac** [reply]
Unfortunately you'll need to wait until they come online and close otherwise there's nothing you can do. I can't connect to those nodes either and if you look in the Amboss channel pages, you'll see that their peers have disabled the channels, meaning they signal that their peer is down.

**Reply by pasq1** [reply]
Damn that stinks okay I appreciate the help, I wasn't sure if there was anything I could do at this point besides wait. So what's the difference between a zombie channel and one like this that can't talk to the other node when I first saw the guide I assumed it was a zombie channel and I would just need to confirm they are offline

**Reply by bitromortac** [reply]
The notion of a zombie channel depends on your state, you can only have zombie channels if you lost your channel state. Otherwise they are just offline peers, which you can force close at anytime, if you have the latest channel.db. It's good practice to force close channels with peers if you notice they are gone for a long time, to decrease the risks you ran into.

**Reply by jimbrend** [reply]
we are able to recover the SCB backup file, and can confirm @pasq1 was able to use it and get out of 8 channels 6 closed (that right?), and then these last 2 are now problematic, 

As opposed to the SCB channel.backup file we are not able to get the channel database, the channels.db file

We can only grab that from the previous installation or possibly previous storage drive if we can still access it 

In regards to the data loss (is like when we are relocating or someone completely losing access to their device or tries to recover elsewhere, or some catastrophic drive failure) obviously that should be avoided, we want to usually close channels or manage them to avoid disaster recovery so question there was what led you to have to recover?  Generally, we can troubleshoot the device to get it working again and avoid that reinstallation for anyone else coming across this,

But can confirm upon reinstallation Umbrel's intended behavior now is it asks you to input your seed and then allows you to select your SCB backup file which broadcasts that close and then returns funds on chain in these disaster situations or someone has lost access to their old device... 

So for these last two channels, @pasq1 does have them in his pendingchannels output as the last two channels, want to confirm if you know of any way @bitromortac that he can't communicate to the network, to my understanding even a remotesweep or maybe something with chantools cannot fix this? 

So the recommendation is @pasq1 should leave this LND instance running anywhere (only one instance at a time with the same seed) and then hopefully these peers come back online and can initiate a cooperative close

**Comment by jimbrend** [comment]
Hey all I wanted to just ping you for visibility @alexbosworth @guggero 

i really appreciate your input as I try to help LND users on Umbrel, we try to troubleshoot devices and avoid the disaster recovery,

Some users still get stuck having only used the SCB file, I know this is not ideal, 

but want to confirm that in this situation there's nothing we can do other then hope @pasq1 's peers come back online for the cooperative close?

His pending channel output is above... appreciate your time whenever available to advise that we must wait for peer to come online and no other intervention here is possible
